src/assets/TokenRatesController.ts

```diff
@@ -312,29 +336,90 @@ export class TokenRatesController extends BaseController<
+      const { nativeCurrency } = this.config;
+      newContractExchangeRates = await this.fetchAndMapExchangeRates(
+        nativeCurrency,
+        slug,
+      );
+    }
+    this.update({ contractExchangeRates: newContractExchangeRates });
+  }
+
+  /**
+   * Checks if the active network's native currency is supported by the coingecko API
+   * If supported. fetches and maps contractExchange rates in format to be consumed by UI
+   * If not supported fetches contractExchange rates and maps from from token/fallback-currency to token/nativeCurrency
+   *
+   * @param nativeCurrency - the native currency of the currently active network
+   * @param slug - the unique slug used to id the chain by the coingecko api
+   * should be used to query token exchange rates
+   * @returns An object with conversion rates for each token
+   * related to the network's native currency
+   */
+  async fetchAndMapExchangeRates(
+    nativeCurrency: string,
+    slug: string,
+  ): Promise<ContractExchangeRates> {
+    const contractExchangeRates: ContractExchangeRates = {};
+
+    // check if native currency is supported as a vs_currency by the API
+    const nativeCurrencySupported = await this.checkIsSupportedVsCurrency(
+      nativeCurrency,
+    );
+
+    if (nativeCurrencySupported) {
+      // if it is we can do a simple fetch against the coingetck API
+      const prices = await this.fetchExchangeRate(slug, nativeCurrency);
       this.tokenList.forEach((token) => {
-        const address = toChecksumHexAddress(token.address);
         const price = prices[token.address.toLowerCase()];
-        newContractExchangeRates[address] = price
+        contractExchangeRates[toChecksumHexAddress(token.address)] = price
           ? price[nativeCurrency.toLowerCase()]
           : 0;
       });
+    } else {
+      // if native currency is not supported we need to use a fallback vsCurrency, get the exchange rates
+      // in token/fallback-currency format and convert them to expected token/nativeCurrency format.
+      let tokenExchangeRates;
+      let nativeCurrencyConversionRate = 0;
+      try {
+        [
+          tokenExchangeRates,
+          { conversionRate: nativeCurrencyConversionRate },
+        ] = await Promise.all([
+          this.fetchExchangeRate(slug, FALL_BACK_VS_CURRENCY),
+          fetchNativeExchangeRate(FALL_BACK_VS_CURRENCY, nativeCurrency, false),
+        ]);
+      } catch (error) {
+        if (
+          error.message.includes('market does not exist for this coin pair')
+        ) {
+          return {};
+        }
+        throw error;
+      }
+
+      for (const [tokenAddress, conversion] of Object.entries(
+        tokenExchangeRates,
+      )) {
+        const fallBackConversionRate =
+          conversion[FALL_BACK_VS_CURRENCY.toLowerCase()];
+        contractExchangeRates[toChecksumHexAddress(tokenAddress)] =
+          nativeCurrencyConversionRate > 0
+            ? fallBackConversionRate / nativeCurrencyConversionRate
```

Comment:
We could avoid the "divide by zero" case here by _multiplying_ by the exchange rate from ETH to the native currency, rather than dividing by the exchange rate from the native currency to ETH. That might be simpler to understand as well - I struggled for a bit trying to understand why we were dividing here.